docs/reference/modules/asset.md

```diff
@@ -0,0 +1,205 @@
+---
+extends: '@apostrophecms/module'
+---
+
+# `@apostrophecms/asset`
+
+**Alias:** `apos.asset`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+The `asset` module serves to organize, process, and output all project script and styling assets during the build process. In addition, it provides access to modify the project webpack configuration and exposes two CLI tasks for project building and webpack cache clearing. Options are passed through the creation of a `modules/@apostrophecms/asset/index.js` file.
+
+## Options
+
+|  Property | Type | Description |
+|---|---|---|
+| [`refreshOnRestart`](#refreshonrestart) | Boolean | If set to `true`, the browser will refresh on Apostrophe app restart. |
+| [`watch`](#watch) | Boolean | If set to `false`, none of the UI assets will be watched for triggering a restart. |
+| `watchDebounceMs` | Integer | Time in milliseconds to wait before re-triggering a restart on asset change. |
+
+### `refreshOnRestart`
+
+If `refreshOnRestart` is truthy and `process.env.NODE_ENV` is not set to `production`, then restart of the Apostrophe app will trigger a browser refresh.
+
+### `watch`
+
+If `watch` is falsy, then changes to any of the assets in the `modules` and `node_modules` will not trigger a restart. Note that this option is disabled and has no impact if `process.env.NODE_ENV` is set to `production`. 
+
+## Command Line Tasks
+
+### `build`
+The build command triggers the compilation, processing, and output of the project assets. The main code from each module, code placed in the `<module-name>/ui/src` folders, and scripts included in [bundles](#bundles) are combined separately. Any SCSS files are combined and translated to CSS. If `NODE_ENV=production`, all of the final deployment files are written out to the `public/releases` folder within a sub-folder named by the release id. The `build` script will try to automatically set the release id when building from a simple git checkout from the hash, building for Heroku or platform.sh from the environment variables, or Stagecoach from the root directory name. Otherwise, the release id can set from a `APOS_RELEASE_ID` environment variable or with a root directory file named 'release-id' with the value of the id inside. The value should be a short, unique string identifier.
```

Comment:
This language is confusing. All of the source code for browserland for non-logged-in users is in ui/src, it's not separate from where bundles live, and there's no "main code from each module" separate from it. Also this task is responsible for the admin UI build too. Please check out my take on this below:



There are two important frontend builds: `ui/apos` and `ui/src`. Both are built by this task. Logged-in users receive both, logged-out users receive the `ui/src` build only.

`ui/apos` handles the Apostrophe admin UI and you can read more about it on the "customizing the admin UI" page. Code for this build is automatically found and incorporated if present in the `ui/apos` subdirectory of every module.

`ui/src` handles frontend JavaScript and SASS CSS for ordinary website visitors. If a module has a `ui/src/index.js` file, that is automatically incorporated as a JavaScript entry point. if a module has a `ui/src/index.scss` file, that is automatically incorporated as a CSS entry point. `ui/src/index.js` files must export a function, and the browser will invoke these in the order the modules were configured.

In addition, files in `ui/public` are automatically concatenated into the JavaScript bundle "as-is."

In addition to the `ui/src/index.js` and `ui/src/index.scss` entry points, additional entry points can be defined by declaring bundles via the `webpack` section of any module's main `index.js` file. These create separately compiled frontend bundles which are loaded only if a widget type or page type module declares a need for them.

Reply:
"or Stagecoach via the timestamp part of the project's deployment folder name."

Reply:
can be set from -> must be set via

Reply:
Changed - I hope I captured your comments correctly.